### PR TITLE
4 connectors

### LINKED ISSUE #3731: [Site Request] mangadig
**Name of the website**
mangadig

**Website urls (examples below)**
- Site: https://mangadig.com/
- Manga List: https://mangadig.com/show?mainCategoryId=10008
- Manga example: https://mangadig.com/10951/
- Chapter online viewer example:  https://mangadig.com/10951/1/381.html

**Languages**
List of languages supported by the website : ENG

**Website relationship*
I get link from cocomanhua.com

### LINKED ISSUE #5281: [https://www.ainzscans.my.id/] Ainz Scans ID
**Name of the website**
Ainz Scans ID

**Website urls (examples below)**
- Site: https://www.ainzscans.my.id/
- Manga List: https://www.ainzscans.my.id/p/manga-list.html
- Manga example: https://www.ainzscans.my.id/2022/12/i-copied-peerless-divine-art-in-another.html
- Chapter online viewer example: https://www.ainzscans.my.id/2022/12/chapter-00.html

**Languages**
Indonesia

**Website relationship*
If applicable describe the relation with any other website (eg : "alternative domain, copy of ...").

**Additional details**
Add any other context details that you may have found (like template to reuse)

### LINKED ISSUE #5287: [Site Request] Manhuaindex.com
**Name of the website**
Manhuaindex

**Website urls (examples below)**
- Site: https://manhuaindex.com/
- Manga List: 
https://manhuaindex.com/series/
- Manga example: https://manhuaindex.com/series/i-really-dont-want-to-remake/

**Languages**
English

### PR DESCRIPTION
**Pull request by MikeZeDev**
Fixes https://github.com/manga-download/hakuneko/issues/5281
Fixes https://github.com/manga-download/hakuneko/issues/5287
Fixes https://github.com/manga-download/hakuneko/issues/3731
other is mangasect

Mangadig is here https://github.com/manga-download/hakuneko-plugins/blob/master/src/MangaDig.js

### CODE REVIEW

File: src/web/mjs/connectors/Ainzscans.mjs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    constructor() {
+        super();
+        super.id = 'ainzscans';
+        super.label = 'Ainz Scans';
+        this.tags = [ 'webtoon', 'indonesian', 'scanlation' ];
+        this.url = 'https://www.ainzscans.my.id';
+    }
+
+    async _getMangas() {
+        const uri = new URL('/feeds/posts/default/-/Series?alt=json&max-results=500', this.url);
```
Comment (Sheepux) [inline src/web/mjs/connectors/Ainzscans.mjs]: if there is a max result then it's missing pagination handling

File: src/web/mjs/connectors/Ainzscans.mjs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        const uri = new URL(manga.id, this.url);
+        const script = `
+          new Promise(resolve => {
+            resolve(label_chapter);
+        });
+        `;
+        let request = new Request(uri);
+        const data = await Engine.Request.fetchUI(request, script, 5000);
+
+        const apiuri = new URL('/feeds/posts/default/-/'+data+'?alt=json&max-results=500', this.url);
```
Comment (Sheepux) [inline src/web/mjs/connectors/Ainzscans.mjs]: if there is a max result then it's missing pagination handling

File: src/web/mjs/connectors/Ainzscans.mjs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        let uri = new URL(chapter.manga.id, this.url);
+        const script = `
+          new Promise(resolve => {
+            resolve(label_chapter);
+        });
+        `;
+        let request = new Request(uri);
+        let data = await Engine.Request.fetchUI(request, script, 5000);
+
+        uri = new URL('/feeds/posts/default/-/'+data+'?alt=json&max-results=500', this.url);
```
Comment (Sheepux) [inline src/web/mjs/connectors/Ainzscans.mjs]: if there is a max result then it's missing pagination handling

File: src/web/mjs/connectors/MangaDig.mjs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+/* eslint-disable */
```
Comment (Sheepux) [inline src/web/mjs/connectors/MangaDig.mjs:1]: line: 46 remove swearing language from code

Comment (MikeZeDev) [reply src/web/mjs/connectors/MangaDig.mjs:1]: Other obfuscated connectors already got similar stuff, but fine i'll do it.

**Review by Sheepux — APPROVED** [review]

**Comment by 27136773** [comment]
- [Missing in HaruNeko](https://github.com/search?q=repo%3Amanga-download%2Fharuneko%20%27ainzscans%27&type=code)
 - [Missing in HaruNeko](https://github.com/search?q=repo%3Amanga-download%2Fharuneko%20%27mangadig%27&type=code)
 - [Missing in HaruNeko](https://github.com/search?q=repo%3Amanga-download%2Fharuneko%20%27mangasect%27&type=code)
 - [Missing in HaruNeko](https://github.com/search?q=repo%3Amanga-download%2Fharuneko%20%27manhuaindex%27&type=code)

**Pull request merged**